contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+    }
+
+    function withdrawETH(address payable to, uint256 amount) external virtual override onlyPCVController {
+        to.transfer(amount);
+    }
+
+    function setBufferWeight(uint128 newBufferWeight) external virtual override onlyGuardianOrGovernor {
+        int128 difference = int128(newBufferWeight) - int128(bufferWeight);
+        bufferWeight = uint128(int128(bufferWeight) + difference);
+
+        totalWeight = uint128(int128(totalWeight) + difference);
+    }
+
+    function setPCVDepositWeight(address depositAddress, uint128 newDepositWeight) external virtual override onlyGuardianOrGovernor {
+        if (!pcvDepositAddresses.contains(depositAddress)) {
+            revert("Deposit does not exist.");
+        }
+
+        int128 difference = int128(newDepositWeight) - int128(pcvDepositInfos[depositAddress].weight);
+        pcvDepositInfos[depositAddress].weight = uint128(newDepositWeight);
+
+        totalWeight = uint128(int128(totalWeight) + difference);
+    }
+
+    function removePCVDeposit(address pcvDeposit) external virtual override onlyGuardianOrGovernor {
+        _removePCVDeposit(address(pcvDeposit));
+    }
+
+    function addPCVDeposit(address newPCVDeposit, uint256 weight) external virtual override onlyGovernor {
+        _addPCVDeposit(address(newPCVDeposit), uint128(weight));
+    }
+
+    function setNewAggregator(address newAggregator) external virtual override onlyGovernor {
+        // Add each pcvDeposit to the new aggregator
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            uint128 pcvDepositWeight = pcvDepositInfos[pcvDepositAddress].weight;
+
+            IPCVDepositAggregator(newAggregator).addPCVDeposit(pcvDepositAddress, pcvDepositWeight);
+        }
+
+        // Set all weights to zero (except for us)
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            pcvDepositInfos[pcvDepositAddress].weight = uint128(0);
+        }
+
+        // Rebalance (withdraws everything to us)
+        _rebalance();
+
+        // Send everything over to the new aggregator
+        IERC20(token).safeTransfer(address(newAggregator), IERC20(token).balanceOf(address(this)));
+
+        // Call rebalance on the new aggregator
+        IPCVDepositAggregator(newAggregator).rebalance();
+
+        // Remove all deposits.
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            _removePCVDeposit(pcvDepositAddress);
+        }
+
+        // Finally, set the new aggregator on the rewards asset manager itself
+        IRewardsAssetManager(rewardsAssetManager).setNewAggregator(address(newAggregator));
+    }
+
+    // ---------- View Functions ---------------
+    function resistantBalanceAndFei() external view virtual override returns (uint256, uint256) {
```

Comment:
This should probably sum resistant balance and FEI of all underlying

Reply:
It does. Well, it does for balance. Should do it for FEI too.